File: app/src/test/java/com/turlir/interpreter/PolishInterpreterTest.java
package com.turlir.interpreter;

import org.junit.Before;
import org.junit.Test;

import java.math.BigDecimal;
import static org.junit.Assert.*;

public class PolishInterpreterTest {

    private NotationInterpreter mInter;

    @Before
    public void setUp() {
        mInter = new PolishInterpreter();
    }

    @Test
    public void singlePoolTest() {
        mInter = new PolishInterpreter();
        mInter.pushDigit(new BigDecimal(2.0));
        BigDecimal actual = mInter.poolDigit();
        assertTrue(new BigDecimal(2.0).compareTo(actual) == 0);
    }

    @Test
    public void multiPollTest() {
        BigDecimal[] array = new BigDecimal[5];
        for (int i = 0; i < array.length; i++) {
            array[i] = new BigDecimal(i);
        }

        mInter = new PolishInterpreter();
        for (BigDecimal digit : array) {
            mInter.pushDigit(digit);
        }

        for (int i = 0; i < array.length; i++) {
            BigDecimal digit = array[array.length - i - 1];
            BigDecimal actual = mInter.poolDigit();
            assertTrue(digit.compareTo(actual) == 0);
        }
    }

}

File: calculator/src/test/kotlin/com/turlir/calculator/interpreter/PolishInterpreterTest.kt
package com.turlir.calculator.interpreter

import com.turlir.calculator.interpreter.NotationInterpreter
import com.turlir.calculator.interpreter.PolishInterpreter
import org.junit.Assert.assertTrue
import org.junit.Test
import java.math.BigDecimal

class PolishInterpreterTest {

    private var mInter: NotationInterpreter = PolishInterpreter()

    @Test
    fun singlePoolTest() {
        mInter = PolishInterpreter()
        mInter.pushDigit(BigDecimal(2.0))
        val actual = mInter.poolDigit()
        assertTrue(BigDecimal(2.0).compareTo(actual) == 0)
    }

    @Test
    fun multiPollTest() {
        val array = (1..5).asSequence()
                .map { BigDecimal(it) }
                .toList()

        mInter = PolishInterpreter()
        for (digit in array) {
            mInter.pushDigit(digit)
        }

        for (i in array.indices) {
            val digit = array[array.size - i - 1]
            val actual = mInter.poolDigit()
            assertTrue(digit.compareTo(actual) == 0)
        }
    }

}